### DISCUSSION TITLE
[ERROR] [backend/drm/atomic.c:79] connector HDMI-A-1: Atomic commit failed: Device or resource busy [ERROR] [../labwc/src/output-state.c:45] Failed to commit frame

### DISCUSSION THREAD

**Discussion by charbelnicolas** [labwc/labwc#2027] [Q&A]
Ever since labwc updated to wlroots 0.18 I've been getting my logs spammed with the following errors:

![ss_2024_07_24_11_54_51](https://github.com/user-attachments/assets/6ddaa14e-dd9f-4848-b7d6-97c6d6f5615a)

Is this something that has to do with my computer/setup or is this fixable from labwc's part?

These errors are just from watching youtube videos on chromium....

**Comment by Consolatis** [comment]
Do you have tearing enabled?

**Comment by charbelnicolas** [comment]
No, this is what I have:

```
<core>

	<decoration>server</decoration>
	<gap>0</gap>
	<adaptiveSync>no</adaptiveSync>
	<allowTearing>no</allowTearing>
	<reuseOutputMode>no</reuseOutputMode>

</core>
```

**Comment by Consolatis** [comment]
@kode54 any clue what the reason for the "device busy error" could be? And if we can do anything about it?
I can't currently run the master branch on DRM due to some missing dependencies on my distro for the wlroots DRM backend.

**Comment by kode54** [comment]
This issue looks relevant:

https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3507

I was experiencing some similar issues with wlroots 0.18 at some point, and I don't know why they went away.

**Comment by charbelnicolas** [comment]
I'm using an intel iGPU too with i915 drivers -_-

**Comment by pablo-888** [comment]
Same (Intel). I just start both sway and labwc with WLR_DRM_NO_ATOMIC=1 to make it go away. I don't notice any drawbacks at all so I just leave it like that. Firefox seems to trigger it the most for me (with Chromium as second) at random times but I don't think it's specific to those and it's just something that happens with resource heavy applications (I remember seeing this a while back https://github.com/labwc/labwc/issues/1507). For example as a silly test I loaded a big sluggish website and while that was loading I immediately opened/closed the gtk filechooser a few times. That seemed to trigger it pretty often.   

(speaking of, does tearing work on intel with atomic modestting enabled for anyone else? It doesn't for me and I remember it didn't work on Hyprland back when they first introduced it either so maybe it's a driver thing). I get this whenever I trigger the keybinding for it: Attempted to submit a tearing page flip to an unsupported backend!

edit: nvm the unsupported backend error doesn't appear anymore (hadn't tested it since the pr was merged) and the logs say it's being enabled and isabled but nothing actually changes.

edit2: my silly test doesn't seem work anymore i can't make the busy error happen on command. however if i enable or disable tearing with the atomic modesetting (which again does nothing visually) a huge wall of OP's errors show up all at once. not sure if that is any helpful at all.

edit3 and last edit before I go to bed and forget about the whole thing: it happened again with tearing disabled and atomic modesetting enabled (not a wall of text this time just a single atomic commit failed error when loading a website). I didn't really have to do anything special this time. It may not be CPU/GPU load dependent like I thought since I tried stressing the browser by reloading a bunch of tabs at the same time while switching between them quickly/scrolling/etc and nothing happened. Idk 😁.

**Reply by charbelnicolas** [reply]
> WLR_DRM_NO_ATOMIC=1

That's what I'm going to do, thank you.

**Reply by charbelnicolas** [reply]
This worked like a charm! No more spam in the error logs :)
